table/block_based/block_based_table_reader.cc

```diff
@@ -93,12 +94,30 @@ class BlocklikeTraits<BlockContents> {
   }
 };
 
+template <>
+class BlocklikeTraits<FullFilterData> {
+ public:
+  static FullFilterData* Create(BlockContents&& contents,
+                                SequenceNumber /* global_seqno */,
+                                size_t /* read_amp_bytes_per_bit */,
+                                Statistics* /* statistics */,
+                                bool /* using_zstd */,
+                                const FilterPolicy* filter_policy) {
+    return new FullFilterData(filter_policy, std::move(contents));
+  }
+
+  static uint32_t GetNumRestarts(const FullFilterData& /* data */) {
+    return 0;
+  }
+};
+
 template <>
 class BlocklikeTraits<Block> {
  public:
   static Block* Create(BlockContents&& contents, SequenceNumber global_seqno,
                        size_t read_amp_bytes_per_bit, Statistics* statistics,
-                       bool /* using_zstd */) {
+                       bool /* using_zstd */,
+                       const FilterPolicy* /* filter_policy */) {
```

Comment:
Extra param is unfortunate, but ...

Actually, I'm not a C++ guru, but is this a place where we could use parameter packs? https://en.cppreference.com/w/cpp/language/parameter_pack

That would also save us passing in null when not needed.

Reply:
Yeah, I think we could make this cleaner by perfect forwarding the ctor arguments. It would need to be done all the way down through the call stack though. I'll take a stab at it.

Reply:
Recommended, but optional

Reply:
As it turns out, that's a bit more involved change; I'll take care of it in a separate patch.